### PR TITLE
Implement Amplication footer

### PR DESCRIPTION
**Pull request by abrl91**
Issue Number: #3577 

## PR Details
Implement Amplication footer:
- git connection status
- last build - only on resource level and should show the last build of the current service
- last commit - on the project + resource level

## PR Checklist
- [ ] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

### CODE REVIEW

**Review by barshimi — APPROVED** [review]

File: packages/amplication-client/src/Resource/git/GitStatusConnectedDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React, { useContext } from "react";
+import { AppContext } from "../../context/appContext";
+
+const CLASS_NAME = "git-status-connected-details";
+
+const GitStatusConnectedDetails: React.FC<{}> = () => {
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Resource/git/GitStatusConnectedDetails.tsx:6]: rename to `GitRepoDetails`

File: packages/amplication-client/src/Resource/git/AppGitStatusPanel.tsx [hunk 2]
```diff
@@ -57,16 +54,14 @@ const AppGitStatusPanel = ({ resource, showDisconnectedMessage }: Props) => {
...
-              <Button
-                buttonStyle={EnumButtonStyle.Text}
-                icon="external_link"
-                eventData={{
-                  eventName: "openGithubCodeView",
-                }}
-              />
-            </a>
+            <GitStatusConnectedDetails />
+            <Button
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Resource/git/AppGitStatusPanel.tsx:58]: missing <a>

File: packages/amplication-client/src/Resource/git/GitActions/RepositoryActions/GithubSyncDetails.tsx [hunk 3]
```diff
@@ -39,18 +38,21 @@ function GithubSyncDetails({
...
   const errorMessage = formatError(disconnectErrorUpdate);
-  const repoUrl = `https://github.com/${gitRepositoryFullName}`;
 
   return (
     <div className={CLASS_NAME}>
       <div className={`${CLASS_NAME}__body`}>
-        <div className={`${CLASS_NAME}__details`}> 
+        <div className={`${CLASS_NAME}__details`}>
           <div className={classNames(className, `${CLASS_NAME}__name`)}>
             {gitRepositoryFullName}
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Resource/git/GitActions/RepositoryActions/GithubSyncDetails.tsx:47]: use <GitRepoDetails/>

File: packages/amplication-client/src/Workspaces/WorkspaceFooter.scss [hunk 4]
```diff
@@ -4,8 +4,75 @@
...
   height: var(--footer-height);
   min-height: var(--footer-height);
+
+  &__skeleton {
+    display: flex;
+    align-items: center;
+  }
+
+  &__vertical_border {
+    width: 1px;
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Workspaces/WorkspaceFooter.scss:19]: try to use :
width:0;
border-right:$border-black60;

File: packages/amplication-client/src/Workspaces/WorkspaceFooter.scss [hunk 5]
```diff
@@ -4,8 +4,75 @@
...
+  &__left {
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+  }
+
+  &__right {
+    .truncated-id {
+      color: var(--black100);
+      font-weight: 400;
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Workspaces/WorkspaceFooter.scss:36]: use mixin

File: packages/amplication-client/src/Workspaces/WorkspaceFooter.tsx [hunk 6]
```diff
@@ -1,10 +1,138 @@
...
+  const handleBuildLinkClick = useCallback((event) => {
+    event.stopPropagation();
+  }, []);
+
+  const ClickableCommitId = lastCommit && (
+    <ClickableId
+      to={`/${currentWorkspace?.id}/${currentProject?.id}/commits/${lastCommit.id}`}
+      id={lastCommit.id}
+      label="Commit ID"
+      onClick={handleBuildLinkClick}
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Workspaces/WorkspaceFooter.tsx:65]: no need for this

File: packages/amplication-client/src/Workspaces/WorkspaceFooter.tsx [hunk 7]
```diff
@@ -1,10 +1,138 @@
...
+      }}
+    />
+  );
+
+  const ClickableBuildId = lastResourceBuild && (
+    <ClickableId
+      label="Build ID"
+      to={`/${currentWorkspace?.id}/${currentProject?.id}/${lastResourceBuild.resourceId}/builds/${lastResourceBuild.id}`}
+      id={lastResourceBuild.id}
+      onClick={handleBuildLinkClick}
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Workspaces/WorkspaceFooter.tsx:77]: no need for this

File: packages/amplication-client/src/Workspaces/hooks/useResources.ts [hunk 8]
```diff
@@ -13,6 +13,16 @@ type TData = {
   createResourceWithEntities: models.Resource;
 };
 
+const createGitRepositoryFullName = (
+  gitRepository: models.Maybe<models.GitRepository> | undefined
+) => {
+  return (
+    (gitRepository &&
+      `${gitRepository.gitOrganization.name}/${gitRepository.name}`) ||
+    "connect to GH"
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Workspaces/hooks/useResources.ts:22]: use full text

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

File: packages/amplication-client/src/Workspaces/WorkspaceFooter.scss (hunk 4 above)
Comment (abrl91) [reply packages/amplication-client/src/Workspaces/WorkspaceFooter.scss:19]: I tried. It doesn't look good.
We also do it like that in WorkspaceHeader
https://github.com/amplication/amplication/blob/master/packages/amplication-client/src/Workspaces/WorkspaceHeader/WorkspaceHeader.scss#L152

**Review by yuval-hazaz — APPROVED** [review]

**Pull request merged**